tests/posix/common/src/pthread.c

```diff
@@ -371,6 +377,67 @@ ZTEST(pthread, test_pthread_termination)
 	zassert_equal(ret, ESRCH, "cancelled a terminated thread!");
 }
 
+ZTEST(pthread, test_pthread_tryjoin)
+{
+	pthread_t th = {0};
+	void *retval;
+
+	/* Creating a threads that exits after 1s*/
+	zassert_ok(pthread_create(&th, NULL, timedjoin_thread, NULL));
```

Comment:
I think it would be nice to pass the sleep time as arg as well:

```suggestion
	useconds_t sleep_time;
	...
	/* Creating a threads that exits after 200ms */
	sleep_time = USEC_PER_MSEC * 200U;
	zassert_ok(pthread_create(&th, NULL, timedjoin_thread, (void *)sleep_time);
```

but probably just personal preference, so non-blocking

Reply:
It's not a bad suggestion, but `INT_TO_POINTER()` and `POINTER_TO_INT()` should be used.

Reply:
Also, we should use "create a thread" instead of "create a threads".

Reply:
I have added this. But with type `int` instead of `useconds_t`, so that I could use the `INT_TO_POINTER`/`POINTER_TO_INT` macros. Let me know what you think.